assets/out

```diff
@@ -130,6 +127,48 @@ current_deployed() {
   $helm_bin history $tls_flag --tiller-namespace $tiller_namespace --max 1 $release | grep "DEPLOYED"
 }
 
+# Function: helm_log_upgrade
+# Description: This function continuously checks the status of pods in a Kubernetes namespace after a Helm upgrade. 
+# It retrieves the pods with a specific label and prints their names and statuses. 
+# If a pod is not in the "Running" state, it prints the pod's logs and events.
+# Parameters:
+#   - release: The name of the Helm release
+#   - namespace: The Kubernetes namespace where the release is deployed
+# This function is meant to be run in the background concurrently with the helm upgrade command
+podLogsDir=/tmp/pod_logs
+helm_log_upgrade() {
+  local helm_pid=$1  
+  mkdir -p $podLogsDir
+  app=$(echo "$release" | cut -d'-' -f1)
+  while kill -0 $helm_pid 2>/dev/null
+  do    
+    pods=$(kubectl -n $namespace get pods -l app=$app --no-headers )
+    while IFS= read -r line; do
+      podName=$(echo $line | awk '{print $1}')
+      podStatus=$(echo $line | awk '{print $3}')    
+      outfile=$podLogsDir/$podName.log
+      rm -f $outfile
```

Comment:
There's a race condition because of this. Since this loop can be terminated at any time due to the `kill` command below, it's possible it could get terminated right after removing the logs. In that case we lose information. Actually, by the same logic it could get terminated with partial logs (and this actually seems pretty likely).

I think the best way to fix this is to replace `wait $helm_pid` below with `wait $helm_log_pid`. This loop should terminate on its own when `helm` finishes so it accomplishes the same thing but allows this function to complete gracefully.

Reply:
Great catch.  ty!

Reply:
You are right that `while kill -0 $helm_pid 2>/dev/null` is esoteric
The kill is redundant as it checks on the condition if helm is killed or not